e2etest/bgp_test.go

```diff
@@ -390,6 +390,130 @@ var _ = ginkgo.Describe("BGP", func() {
 			}),
 		)
 	})
+
+	ginkgo.Context("validate configuration changes", func() {
+		ginkgo.AfterEach(func() {
+			// Clean previous configuration.
+			err := updateConfigMap(cs, configFile{})
+			framework.ExpectNoError(err)
+		})
+
+		ginkgo.It("request a specific address pool for each Loadbalancer service", func() {
+			var peers []peer
+			var services [2]*corev1.Service
+			var servicesIngressIP [2]string
+			var pools []addressPool
+
+			namspace := f.Namespace.Name
+
+			for _, c := range frrContainers {
+				peers = append(peers, peer{
+					Addr:     c.Ipv4,
+					ASN:      c.RouterConfig.ASN,
+					MyASN:    c.NeighborConfig.ASN,
+					Port:     c.RouterConfig.BGPPort,
+					RouterID: c.RouterConfig.RouterID,
+					Password: c.RouterConfig.Password,
+					HoldTime: c.RouterConfig.HoldTime,
+				})
+			}
+
+			allNodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+			framework.ExpectNoError(err)
+
+			for i := 0; i < 2; i++ {
+				ginkgo.By(fmt.Sprintf("configure addresspool number %d", i+1))
+				firstIP := getIPFromRangeByIndex("192.168.10.0/24", i*10+1)
+				lastIP := getIPFromRangeByIndex("192.168.10.0/24", i*10+10)
+				addressesRange := fmt.Sprintf("%s-%s", firstIP, lastIP)
+				pool := addressPool{
+					Name:     fmt.Sprintf("test-addresspool%d", i+1),
+					Protocol: BGP,
+					Addresses: []string{
+						addressesRange,
+					},
+				}
+				pools = append(pools, pool)
+
+				configData := configFile{
+					Pools: pools,
+					Peers: peers,
+				}
+				err := updateConfigMap(cs, configData)
+				framework.ExpectNoError(err)
+
+				for _, c := range frrContainers {
+					pairExternalFRRWithNodes(cs, c)
+				}
+
+				ginkgo.By(fmt.Sprintf("configure service number %d", i+1))
+				jig := e2eservice.NewTestJig(cs, namspace, fmt.Sprintf("test-service%d", i+1))
```

Comment:
Is there a reason why you can't reuse `createServiceWithBackend` ?

Reply:
i didn't want to add a name parameter to this function. do you think we should?

Reply:
That doesn't sound too bad. Another option is to add a slice of optional modifier functions so all the other calling sites won't be affected. Something like 
`createServiceWithBackend(cs clientset.Interface, namespace string, policy corev1.ServiceExternalTrafficPolicyType, modifiers ...func(*corev1.Service))

We can even refactor the traffic policy parameter to use that.
See https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis for more details.

Either ways are fine, right now it's a bit hard to follow.